src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -23,8 +27,26 @@ namespace Microsoft.Build.Logging.TerminalLogger;
 /// <remarks>
 /// Uses ANSI/VT100 control codes to erase and overwrite lines as the build is progressing.
 /// </remarks>
-internal sealed class TerminalLogger : INodeLogger
+internal sealed partial class TerminalLogger : INodeLogger
 {
+    private const string FilePathPattern = " -> ";
+
+#if NET7_0_OR_GREATER
+    [StringSyntax(StringSyntaxAttribute.Regex)]
+#endif
+    private const string ImmediateMessagePattern = @"\[CredentialProvider\]|--interactive";
+
+    private const RegexOptions Options = RegexOptions.IgnoreCase | RegexOptions.CultureInvariant | RegexOptions.ExplicitCapture;
+
+#if NET7_0_OR_GREATER
+    [GeneratedRegex(ImmediateMessagePattern, Options)]
+    private static partial Regex ImmediateMessageRegex();
+#else
+    private static Regex ImmediateMessageRegex() => immediateMessageRegex;
+
+    private static readonly Regex immediateMessageRegex = new(ImmediateMessagePattern, RegexOptions.Compiled | Options);
```

Comment:
Regex matching on .NET Framework tends to be much slower than on .NET, even with `RegexOptions.Compiled`. For the particular pattern used here I see a pair of `IndexOf` calls being about 2x faster than the regex (on amd64). Please consider keeping the regex only for NET7+.